scripts/python/phdf.py

```diff
@@ -109,26 +109,23 @@ def __init__(self, filename):
             self.TotalCells = self.NumBlocks * self.CellsPerBlock
 
             # Read in coordinates
-            tmpx = f['/Locations/x'][:,:]
-            self.x = np.zeros((self.NumBlocks,self.MeshBlockSize[0]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[0]):
-                    self.x[bId,cId] = 0.5*(tmpx[bId,cId]+tmpx[bId,cId+1])
-            self.xf = tmpx
-
-            tmpy = f['/Locations/y'][:,:]
-            self.y = np.zeros((self.NumBlocks,self.MeshBlockSize[1]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[1]):
-                    self.y[bId,cId] = 0.5*(tmpy[bId,cId]+tmpy[bId,cId+1])
-            self.yf = tmpy
-
-            tmpz = f['/Locations/z'][:,:]
-            self.z = np.zeros((self.NumBlocks,self.MeshBlockSize[2]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[2]):
-                    self.z[bId,cId] = 0.5*(tmpz[bId,cId]+tmpz[bId,cId+1])
-            self.zf = tmpz
+            def load_coord(coord_i):
+                coord_name = ["x","y","z"][coord_i]
+
+                tmp = f[f'/Locations/{coord_name}'][:,:]
+                if f'/VolumeLocations/{coord_name}' in f:
+                    coord = f[f'/VolumeLocations/{coord_name}'][:,:]
```

Comment:
Hmm, the fact that `f` is used as the file and as the prefix for formatting strings is a bit confusing here... Can we maybe do this?
```suggestion
                tmp = f['/Locations/' + coord_name][:,:]
                vol_loc = '/VolumeLocations/' + coord_name
                if vol_loc in f:
                    coord = f[vol_loc][:,:]
```

Reply:
That's fine here. Usually I avoid concatenations with strings and prefer the python string formatting, but here it is confusing with the file `f`.